### DISCUSSION TITLE
setValue a useFieldArray inside useEffect is not possible?

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#5131] [Q&A]
Using setValue with useFieldArray is not possible inside useEffect? The fields always changes every time I re-render the page. What I want is to get all the previous data and not set a new value for my fields(useFieldArray) 

I noticed this with the id of the array it changes everytime my page re-render. Any tips on how to not change the previous field inside useEffect

**Comment by JamesWong1999** [comment]
provide a codesandbox.
